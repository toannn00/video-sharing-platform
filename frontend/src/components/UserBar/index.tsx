import { Button } from "antd";
import User from "../../types/user.type";
import { useNavigate } from "react-router-dom";

const UserBar = ({ email }: Partial<User>) => {
  const navigate = useNavigate();

  return (
    <div
      style={{
        display: "flex",
        alignItems: "center",
        gap: 10,
        margin: "0 20px",
      }}
    >
      <div>
        Welcome <b>{email}</b>{" "}
      </div>
      <Button onClick={() => navigate("/post")}>Post</Button>
    </div>
  );
};

export default UserBar;
