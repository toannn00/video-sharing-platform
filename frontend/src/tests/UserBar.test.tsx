import "@testing-library/jest-dom";
import ShallowRenderer from "react-test-renderer/shallow";
import UserBar from "../components/UserBar";
import { vi } from "vitest";

vi.mock("react-router-dom", () => ({
  useNavigate: () => vi.fn(),
}));

describe("UserBar", () => {
  test("should render and match the snapshot", () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<UserBar email="test@example.com" />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
