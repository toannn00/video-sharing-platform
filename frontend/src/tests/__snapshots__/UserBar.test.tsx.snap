// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserBar > should render and match the snapshot 1`] = `
<div
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "gap": 10,
      "margin": "0 20px",
    }
  }
>
  <div>
    Welcome 
    <b>
      test@example.com
    </b>
     
  </div>
  <Button
    onClick={[Function]}
  >
    Post
  </Button>
</div>
`;
